### PR TITLE
Top-Level Resources

### PR DESCRIPTION
**Pull request by gasi**
TJ,

First of all, thanks for all your amazing contributions to the Node.js community.
I've added support for top-level resources to the express-resource module as well as the corresponding tests and documentation. Hope you find this useful.

Cheers,
Daniel

### CODE REVIEW

**Comment by tj** [comment]
cool man :D I will take a look shortly

File: History.md [hunk 1]
```diff
@@ -1,4 +1,9 @@
 
+0.0.3 / 2011-03-27
+==================
+
+  * Added support for top-level resources (using the empty string as name)
```
Comment (tj) [inline History.md:5]: I'll handle releases :p

File: Readme.md [hunk 2]
```diff
@@ -67,6 +67,26 @@ Actions are then mapped as follows (by default):
...
     DELETE  /forums/:id       ->  destroy
 
+
+Specify a top-level resource using the empty string:
+
+    var express = require('express')
+      , Resource = require('express-resource')
+      , app = express.createServer();
+
+    app.resource('', require('./forum'));
```
Comment (tj) [inline Readme.md:77]: it would be cool to just have `app.resource(obj)`, I'll update that

**Comment by tj** [comment]
merged / updated

File: History.md (hunk 1 above)
Comment (gasi) [inline History.md:5]: Sorry, my bad, Release-Master ;)

**Comment by gasi** [comment]
Nice job -- I like the API improvements you've made. I'm officially no longer a pull-request virgin ;)

File: History.md (hunk 1 above)
Comment (tj) [inline History.md:5]: :D

Comment (gasi) [inline History.md:5]: My friend, @aseemk, and I were wondering how you so elegantly integrated my changes with yours. What technique do you use to have me listed as author and you as committer?

Comment (tj) [inline History.md:5]: all you have to do is `curl https://github.com/visionmedia/express-resource/pull/1.patch | git am --signoff`, however I added a command today actually that does this for you to my git-extras repo:

https://github.com/visionmedia/git-extras

with that installed all you do is `git pull-request 1`

Comment (gasi) [inline History.md:5]: TJ, thanks for the explanation. I'll definitely check out your git-extras tools :)

**Pull request closed without merging**